### PR TITLE
feat: Add MockChatService to ChatGPT

### PR DESCRIPTION
**Pull request by eriklimakc**

### CODE REVIEW

File: UI/ChatGPT/ChatGPT/ChatGPT/appsettings.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "AppConfig": {
     "Environment": "Production",
-    "ApiKey": "<Insert your OpenAI API key here. Access https://platform.openai.com/api-keys to generate your OpenAI API key.>"
+    "ApiKey": "" //<Insert your OpenAI API key here. Access https://platform.openai.com/api-keys to generate your OpenAI API key.>
```
Comment (dr1rrb) [inline UI/ChatGPT/ChatGPT/ChatGPT/appsettings.json:4]: Json spec does not define a syntax for comments in json. It's commonly accepted to have `//` but this is not valid regarding to the json spec. Did you validated that it works on all platforms?

File: UI/ChatGPT/ChatGPT/ChatGPT/Business/Models/ChatAiOptions.cs [hunk 2]
```diff
@@ -6,6 +6,6 @@ public class ChatAiOptions: OpenAiOptions
 {
 	public ChatAiOptions(IOptions<AppConfig> appConfig)
 	{
-		this.ApiKey = appConfig.Value.ApiKey ?? throw new InvalidOperationException("You must define an OpenAI API key in appsettings.json file.");
+		this.ApiKey = appConfig.Value.ApiKey!;
```
Comment (dr1rrb) [inline UI/ChatGPT/ChatGPT/ChatGPT/Business/Models/ChatAiOptions.cs:9]: I would not remove the null check here. If the app uses the mock, this `ChatAiOptions` should not be resolved from the DI so it won't throw any exception (we could even imagine to not register it at all when using mock).

File: UI/ChatGPT/ChatGPT/ChatGPT/App.cs [hunk 3]
```diff
@@ -53,12 +53,22 @@ protected override void OnLaunched(LaunchActivatedEventArgs args)
...
-					services
-						.AddSingleton<OpenAiOptions, ChatAiOptions>()
-						.AddSingleton<IChatCompletionService, OpenAIService>()
-						.AddSingleton<IChatService, ChatService>()
-						.AddSingleton<BindableMainModel>())
+				.ConfigureServices(
+					(context, services) => {
+
+						var apiKey = context.Configuration.GetSection("AppConfig:ApiKey").Value;
+						bool useMockService = apiKey is null or { Length: 0 };
```
Comment (dr1rrb) [inline UI/ChatGPT/ChatGPT/ChatGPT/App.cs:60]:
```suggestion
						var useMockService = apiKey is null or { Length: 0 };
```

File: UI/ChatGPT/ChatGPT/ChatGPT/App.cs [hunk 4]
```diff
@@ -53,12 +53,22 @@ protected override void OnLaunched(LaunchActivatedEventArgs args)
...
+
+						services
+							.AddSingleton<OpenAiOptions, ChatAiOptions>()
+							.AddSingleton<IChatCompletionService, OpenAIService>()
+							.AddSingleton<BindableMainModel>();
+
+						if (useMockService)
+							services.AddSingleton<IChatService, MockChatService>();
+						else
+							services.AddSingleton<IChatService, ChatService>();
```
Comment (dr1rrb) [inline UI/ChatGPT/ChatGPT/ChatGPT/App.cs:70]:
```suggestion
						if (useMockService)
						{
							services.AddSingleton<IChatService, MockChatService>();
						}
						else
						{
							services.AddSingleton<IChatService, ChatService>();
						}
```

File: UI/ChatGPT/ChatGPT/ChatGPT/Business/Models/ChatAiOptions.cs (hunk 2 above)
Comment (eriklimakc) [reply UI/ChatGPT/ChatGPT/ChatGPT/Business/Models/ChatAiOptions.cs:9]: If we don't register `ChatAiOptions` when using mock, we can consider not registering `OpenAIService` as well. Then we will have something like:

```
.ConfigureServices(
	(context, services) => {

		var apiKey = context.Configuration.GetSection("AppConfig:ApiKey").Value;
		var useMockService = apiKey is null or { Length: 0 };

		if (useMockService)
		{
			services.AddSingleton<IChatService, MockChatService>();
		}
		else
		{
			services
				.AddSingleton<IChatCompletionService, OpenAIService>()
				.AddSingleton<IChatService, ChatService>()
				.AddSingleton<OpenAiOptions, ChatAiOptions>(); 
		}
		
		services.AddSingleton<BindableMainModel>();
	})
```

What do you think?

File: UI/ChatGPT/ChatGPT/ChatGPT/appsettings.json (hunk 1 above)
Comment (eriklimakc) [reply UI/ChatGPT/ChatGPT/ChatGPT/appsettings.json:4]: Hmmm... Yeah, it worked on Windows, Skia.GTK, WASM and Android. I'm unable to test iOS at the moment.

File: UI/ChatGPT/ChatGPT/ChatGPT/Business/Models/ChatAiOptions.cs (hunk 2 above)
Comment (dr1rrb) [reply UI/ChatGPT/ChatGPT/ChatGPT/Business/Models/ChatAiOptions.cs:9]: Yes, if not used it's better that way 👍

File: UI/ChatGPT/ChatGPT/ChatGPT/App.cs (hunk 4 above)
Comment (eriklimakc) [reply UI/ChatGPT/ChatGPT/ChatGPT/App.cs]: Done in next commit.

File: UI/ChatGPT/ChatGPT/ChatGPT/App.cs (hunk 3 above)
Comment (eriklimakc) [reply UI/ChatGPT/ChatGPT/ChatGPT/App.cs]: Done in next commit.

**Review by nickrandolph — APPROVED** [review]

File: UI/ChatGPT/ChatGPT/ChatGPT/App.cs [hunk 5]
```diff
@@ -53,12 +53,26 @@ protected override void OnLaunched(LaunchActivatedEventArgs args)
...
-				.ConfigureServices(services =>
-					services
-						.AddSingleton<OpenAiOptions, ChatAiOptions>()
-						.AddSingleton<IChatCompletionService, OpenAIService>()
-						.AddSingleton<IChatService, ChatService>()
-						.AddSingleton<BindableMainModel>())
+				.ConfigureServices(
+					(context, services) =>
+					{
+						var apiKey = context.Configuration.GetSection("AppConfig:ApiKey").Value;
```
[the remark this answers is not in the document]
Comment (eriklimakc) [reply UI/ChatGPT/ChatGPT/ChatGPT/App.cs]: Done in next commit.

**Review by nickrandolph — APPROVED** [review]

**Pull request merged**